docker/test_env/besu_test.go

```diff
@@ -0,0 +1,105 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+)
+
+func TestEth2WithPrysmAndBesu(t *testing.T) {
+	l := logging.GetTestLogger(t)
+
+	builder := NewEthereumNetworkBuilder()
+	cfg, err := builder.
+		WithConsensusType(ConsensusType_PoS).
+		WithCustomNetworkParticipants([]EthereumNetworkParticipant{
+			{
+				ConsensusLayer: ConsensusLayer_Prysm,
+				ExecutionLayer: ExecutionLayer_Besu,
+				Count:          1,
+			},
+		}).
+		WithoutWaitingForFinalization().
+		Build()
+	require.NoError(t, err, "Builder validation failed")
+
+	_, eth2, err := cfg.Start()
+	require.NoError(t, err, "Couldn't start PoS network")
+
+	nonEip1559Network := blockchain.SimulatedEVMNetwork
+	nonEip1559Network.Name = "Simulated Besu + Prysm (non-EIP 1559)"
+	nonEip1559Network.GasEstimationBuffer = 10_000_000_000
+	nonEip1559Network.URLs = eth2.PublicWsUrls()
+	clientOne, err := blockchain.ConnectEVMClient(nonEip1559Network, l)
+	require.NoError(t, err, "Couldn't connect to the evm client")
+
+	defer func() {
```

Comment:
Why not use [t.Cleanup()](https://ieftimov.com/posts/testing-in-go-clean-tests-using-t-cleanup/)?

Reply:
true, hasn't occurred to me!